docs/pbc/all/tax-management/202311.0/base-shop/third-party-integrations/vertex/install-vertex.md

```diff
@@ -136,25 +140,67 @@ use Spryker\Zed\TaxApp\Communication\Plugin\Calculation\TaxAppCalculationPlugin;
 
-Spryker has its own [Taxes](/docs/pbc/all/tax-management/{{page.version}}/base-shop/tax-feature-overview.html) feature, which comes pre-installed in the Checkout through the Calculation module. To enhance performance when using an external Tax calculation provider, we recommend disabling the following plugins:
+Please pay attention to the position of TaxAppCalculationPlugin in the list of plugins.
 
-in `\Pyz\Zed\Calculation\CalculationDependencyProvider::getQuoteCalculatorPluginStack()`:
+{% endinfo_block %}
 
-- TaxAmountCalculatorPlugin
-- ItemTaxAmountFullAggregatorPlugin
-- TaxRateAverageAggregatorPlugin
-- TaxTotalCalculatorPlugin
+Update `src/Pyz/Zed/TaxApp/TaxAppDependencyProvider.php` as follows:
 
-in `\Pyz\Zed\Calculation\CalculationDependencyProvider::getOrderCalculatorPluginStack()`:
+```php
+// ...
 
-- TaxAmountCalculatorPlugin
-- ItemTaxAmountFullAggregatorPlugin
-- TaxAmountAfterCancellationCalculatorPlugin
-- OrderTaxTotalCalculationPlugin
+use ;
 
-Disabling them will also disable Spryker Taxes feature. This means that in case Vertex is unresponsive or disabled, order taxes will not be calculated, and their amount will always be 0.
+// ...
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return array<\Spryker\Zed\CalculationExtension\Dependency\Plugin\CalculationPluginInterface>
+     */
+    protected function getFallbackQuoteCalculationPlugins(): array
+    {
+        return [
+            # These plugins will be called if TaxApp configuration is missing or TaxApp is disabled.
+            # Please note that this list includes PriceToPayAggregatorPlugin - this plugin isn't a part of tax calculation logic but it is required by TaxRateAverageAggregatorPlugin.
+            new TaxAmountCalculatorPlugin(),
+            new ItemTaxAmountFullAggregatorPlugin(),
+            new PriceToPayAggregatorPlugin(),
+            new TaxRateAverageAggregatorPlugin(),
+        ];
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return array<\Spryker\Zed\CalculationExtension\Dependency\Plugin\CalculationPluginInterface>
+     */
+    protected function getFallbackOrderCalculationPlugins(): array
+    {
+        return [
+            # These plugins will be called if TaxApp configuration is missing or TaxApp is disabled.
+            # Please note that this list includes PriceToPayAggregatorPlugin - this plugin isn't a part of tax calculation logic but it is required by TaxAmountAfterCancellationCalculatorPlugin.
+            new TaxAmountCalculatorPlugin(),
+            new ItemTaxAmountFullAggregatorPlugin(),
+            new PriceToPayAggregatorPlugin(),
+            new TaxAmountAfterCancellationCalculatorPlugin(),
+        ];
+    }
+    
+// ...
+```
+
+In general, `getFallback<Quote|Order>CalculationPlugins` methods should contain the tax calculation plugins and all required plugins which are replaced by TaxAppCalculationPlugin in `\Pyz\Zed\Calculation\CalculationDependencyProvider`.
```

Comment:
this statement is not fully clear to me.
1. Please use `getFallbackQuoteCalculationPlugins()` and `getFallbackOrderCalculationPlugins()`.
2. `all required plugins which are replaced by TaxAppCalculationPlugin` The customers don't need to know the logic of this plugin, so we need to specify what plugins are needed or NOT needed and in what scenarios.

Reply:
as for 2: the idea here is that we ask the customer to move all the tax-related plugins (except for TaxTotal) into the "fallback" stacks
we can list the OOTB tax plugins we expect them to move but there might be other plugins they have there, right?

Reply:
they might have, but the change is pretty low, so I would list all needed OOTB Spryker plugins (BTW please change OOTB to full name :) ). Let's don't mention custom plugins there at all, I expect that if project decides to create calculation plugins they understand how to deal with them and debug them.